src/os/bluestore/BlueFS.cc

```diff
@@ -3813,6 +3813,13 @@ int BlueFS::truncate(FileWriter *h, uint64_t offset)/*_WF_L*/
     auto p = fnode.seek(offset, &x_off);
     if (p != fnode.extents.end()) {
       uint64_t cut_off = p2roundup(x_off, alloc_size[p->bdev]);
+      if (cut_off > p->length) {
+        // extent was allocated from shared, on fallback to smaller AU
```

Comment:
With new 4K BlueFS min alloc unit we can get here for non-shared devs as well. E.g. when user downsized bluefs alloc size in the past but has reveerted that change since then. Or when we're dealing with that "compatibility" mode when dedicated allocations aren't properlu aligned for historic reasons.